File: validation/poc-00/java-candidate/src/main/java/dev/vra/poc00/domain/OrderItem.java
package dev.vra.poc00.domain;

import java.util.Objects;

public record OrderItem(SkuId skuId, String productNameSnapshot, int quantity, Money unitPrice) {
    public OrderItem {
        Objects.requireNonNull(skuId, "skuId");
        Objects.requireNonNull(productNameSnapshot, "productNameSnapshot");
        Objects.requireNonNull(unitPrice, "unitPrice");
        if (productNameSnapshot.isBlank()) throw new IllegalArgumentException("Product name is required.");
        InventoryBalance.requireQuantity(quantity);
    }
}

File: validation/poc-00/kotlin-candidate/src/main/kotlin/dev/vra/poc00/kotlin/domain/OrderItem.kt
package dev.vra.poc00.kotlin.domain

data class OrderItem(
    val skuId: SkuId,
    val productNameSnapshot: String,
    val quantity: Int,
    val unitPrice: Money
) {
    init {
        require(productNameSnapshot.isNotBlank()) { "Product name is required." }
        InventoryBalance.requireQuantity(quantity)
    }
}
